### PR TITLE
Security panel rewrite

### PR DESCRIPTION
**Pull request by Hexagon**
Submitting for review. Pretty much done by now.

**Background:** 

http://domoticz.com/forum/viewtopic.php?f=31&t=21267 . 

This work covers point 1 and 2, and prepares for 3.

It also covers point a, c and d of this post http://domoticz.com/forum/viewtopic.php?f=31&t=21267#p165551

**Fixed issues:** 
#2040, #1932

### CODE REVIEW

File: www/secpanel/index.html [hunk 1]
```diff
@@ -1,422 +1,157 @@
...
 		
+		// Setup translation
 		function SetLanguageEx(lng) {
+			
 			$.i18n.init({
 				resGetPath: '/i18n/domoticz-__lng__.json',
 				fallbackLng: false,
 				getAsync: false,
-				debug: false,
+				debug: true,
```
Comment (gizmocuz) [inline www/secpanel/index.html]: Could you set this to false ?

**Comment by gizmocuz** [comment]
Thanks for the PR, some comments requests:
- you enabled debugging for I18n in the code
- no need to create a separate JS file for the secpanel code, is/was find in the html
- no need for theme support, please keep it 1 look as it was
- you can not make changes in domoticz-en.json.gz, please tell in the pull request the needed additions so i can add them in our transifex translation system. Also try to avoid additions/changes as everybody needs to translate them
- i see a new file 'ion.sound.min.js', please compress it to ion.sound.min.js.gz
- please post some screenshots, there is a new font ? was this needed?

**Comment by Hexagon** [comment]
Thanks for the review, I'll sort out the points you listed tonight.

Guess we could settle for one theme, the reason i added two more was to open for user selectable themes in a near future. OOTB the good old theme is the default, and isn't interchangeable for normal users.

The light theme looks good on an ipad (white) while the dark theme blends well on a black android tab.

There is no new fonts, i just moved them to the new theme directory.

Screenshots as requested:

Light theme
Light theme

**Comment by gizmocuz** [comment]
Thanks for the screenshots! I would settle for the dark theme (as it also looks better on my 'dark' ipad ;)
But maybe make the buttons/lcd a bit brigher ? it is a bit to dark, also the tekst 'Domoticz security panel' is a bit to dark, that could be normal white with a black shadow/border ?
From the screenshot i can nor make out if i am allowed to press buttons.. they all look disabled.. i guess the 0-9 keys i should be able to press now ? Maybe make them in a color (brighness) that indicates i am allowed to enter a code.
When the code is entered (a code) then the function keys should be allowed to press (also a color thing)

**Comment by Hexagon** [comment]
Great feedback, will have a look at that tonight!

**Comment by Hexagon** [comment]
All above fixed.

Translation changes:

* Changed english translation of "Arm Away" to "Away", kept original reference
* Changed english translation of "Arm Home" to "Home", kept original reference
* Added translation of
   * "Arm Delay"
   * "Disconnected"
   * "Armed Home"
   * "Armed"
   * "Unknown Error" 

```diff
--- domoticz-en-orig.json       2018-01-19 22:21:04.526480550 +0100
+++ domoticz-en.json    2018-01-19 21:23:37.745143194 +0100
@@ -392,10 +392,15 @@
     "No Motion": "No Motion",
     "Panic": "Panic",
     "Panic End": "Panic End",
-    "Arm Away": "Arm Away",
+    "Arm Away": "Away",
     "Arm Away Delayed": "Arm Away Delayed",
-    "Arm Home": "Arm Home",
+    "Arm Home": "Home",
     "Arm Home Delayed": "Arm Home Delayed",
+    "Arm Delay": "Arm Delay",
+    "Disconnected": "Disconnected",
+    "Armed Home": "Armed Home",
+    "Armed": "Armed",
+    "Unknown Error": "Unknown Error",
     "Disarm": "Disarm",
     "Light Off": "Light Off",
     "Light On": "Light On",
```
Screenshot of brighter theme:

Darker dark theme

**Comment by flatsiedatsie** [comment]
Please do add theming support!

**Comment by gizmocuz** [comment]
Is it possible you could keep the original labels ? (Like Arm Home), they are based on current (real) hardware that also use these names. It is confusing to see here 'Away' and in the gui 'Arm Away', we also do not need to do any translations

**Comment by gizmocuz** [comment]
Could you undo the translations ?

**Comment by xztraz** [comment]
Button text should be translatable i think. or was the translation just weird? localisation should be everywhere.

**Comment by gizmocuz** [comment]
The buttons where labeled correctly, they are now renamed, and all languages would need to be updated, also internally.
Best to leave the label names as there where

**Comment by gizmocuz** [comment]
Is it possible to revert the labels ? Else i have to close this patch unfortunally

**Comment by Hexagon** [comment]
I have no spare time at the moment, so you can close it. I'll rebase the changes on top of current development when time pops up.

**Comment by gizmocuz** [comment]
Thanks! Looking forward for it, looks very nice!

**Comment by gizmocuz** [comment]
Thanks! Going to make some tiny language modifications

**Comment by furriephillips** [comment]
This PR is being eagerly & until now, patiently awaited, by at least a few people; is there any news on the status & if we can help get it over the line?

**Comment by gizmocuz** [comment]
From what i see this is patch is already merged ? That is why this PR is closed

**Comment by furriephillips** [comment]
From a UI/UX perspective, although the current Keypad has a darker theme, it seems less than responsive (it appears to be >100% width of my phone’s screen, and cannot be changed/zoomed), and the promised additional * & # buttons are not there.

**Comment by gizmocuz** [comment]
I see, ok then another PR needs to be made. This PR is closed

**Pull request merged**